### ISSUE TITLE
Allow overriding the output file for a target

### ISSUE THREAD

**Issue by peacememories** [lydell/elm-watch#68]
Hi, and thanks for the great work you're doing on elm-watch. We're using it happily in our Django application at work. There are a few features that would make life much easier for us, especially as we would like to integrate our build more directly into Django.

One of them would be to allow either printing the build input for (single target) builds to stdout, or override the target location from the commandline. This would help us dynamically call the build for a module while rendering a Django page in development mode, so we would not have to depend on a separate process running. We would still spawn the separate process for `elm-watch hot` when developing, but this way if you forgot you started it you'd just get a non-hot-reload version instead of an outdated build or a missing js file.

**Comment by lydell** [comment]
Hi!

I don’t quite understand what you’re trying to do. Could you provide a lot more detail? Feel free to talk about this and #69 here, I think that’ll be easier than having two conversations going on at the same time. I used to work with Django a couple of years ago, but I’m a bit rusty.

**Comment by lydell** [comment]
TIL about [docker compose watch](https://docs.docker.com/compose/file-watch/) – maybe that can help in your situation, not sure.

**Comment by peacememories** [comment]
Sorry, I completely forgot that you posted an answer here - sadly, docker compose watch would probably not help for us for two reasons:
1. Rebuilding the container takes too long to make the feedback loop viable
2. We actually want to move away from relying on docker compose in the long term, and are moving to Arion (a nix wrapper of docker compose) in the short term.

> FYI, a goal of elm-watch is that you should never need to install it. I have a bunch of code for dealing with elm crashing, and I even watch the elm-stuff/ folder being deleted and cause a re-compile then. So I’m sad to hear stuff get messed up for you where you need to restart 😞

The main reason this happened for us is when some part of elm/docker/vscode broke ownership in the elm-stuff folder because of the bind mount, and the elm builds started failing. This had to be manually fixed, and elm-watch had to be restarted there. I wouldn't beat myself up about this, overall the resilience of elm-watch is leagues ahead of any other tool I'm aware of.

> At work, all of us are fullstack (more or less – some are leaning more towards backend, and some more towards frontend). Everyone runs one command to start all backend stuff, and in another terminal tab we run another command to start all frontend stuff (including elm-watch). So until now I haven’t considered the case where there’s backend-only people who refuse to install/start frontend stuff.

One thing I forgot that makes this even more complicated for us is that not only are there people not aware/willing to engage with the frontend pipeline, the same development stack might also be used for completely different applications (it's a very janky multiservice application), which depend on some of the frontend in some cases, but are in completely different repositories 😅 we really know how to construct a torment nexus here.

> What I would try, is probably writing a Django thing that compiles on request, like you say. I would probably do that in Python and not involve elm-watch. You could read the elm-watch.json file to tell which input files you need to compile to create the requested output file. Then call elm make directly, and not bother with elm-watch make. To make it extra robust, make sure in your Python code that you don’t call elm make in parallel – that can cause errors during the Elm packages installation phase.

You're probably right... I'm not super happy with it, because that means you need to start the servers differently depending on whether you want to work on the frontend or not, but it's also better in the sense that you then don't have a hot reload build trying to find a nonexistant hot reload server and showing the failing status in the corner.

Maybe I can do some trickery to automatically disable on-render compiling when the file is new enough or something weird like that 😅 

Thanks for taking the time to look into this and for your suggestions :)

**Comment by lydell** [comment]
Thank you for the thorough answers, and good luck on your setup!

I’m going to close these two issues for now, but feel free to comment more if you find something new and we can re-open or create a new issue in the future.
